plugins/module_utils/network/eos/argspec/interfaces/interfaces.py

```diff
@@ -50,6 +50,11 @@ def __init__(self, **kwargs):
                 "mtu": {"required": False, "type": "int"},
                 "speed": {"required": False, "type": "str"},
                 "duplex": {"required": False, "type": "str"},
+                "mode": {
+                    "choices": ["layer2", "layer3"],
+                    "type": "str",
+                    "default": "layer2",
```

Comment:
Setting a default here inadvertently changes the mode of the interfaces to L2 even in a `merged` task. For example, if the interface is already at L3 and we run a task which just sets the description of the interface, it ends up changing the mode to L2, which is unexpected.

Reply:
Existing config:
```
interface Ethernet1
   no switchport
```
Task:
```yaml
 - eos_interfaces:
     config:
      - name: Ethernet1
        description: TEST
```
Commands:
```
 "interface Ethernet1",
  "switchport",
  "description TEST"
```

Reply:
You are right. I have removed the default value for 'mode'. 'mode' will defaulted to 'layer2', whenever the interface is cleared/defaulted.

Reply:
@GomathiselviS just for my understanding, is the default state always layer2 for all EOS versions? Can the system defaults be changed?

Reply:
From what I checked last, default mode of an interface on an eos device is switched mode (layer2).

Reply:
Alright.